internal/http/server.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+const ShutdownGracePeriod = 5 * time.Second
+
+type Server struct {
+	listen *url.URL
+	Server *http.Server
```

Comment:
We can probably embed this directly.

Reply:
DUUUUUUDE!!! Thanks you. This is so much cleaner. Sorry it took me a bit to get there 😂